src/lightning/fabric/wrappers.py

```diff
@@ -132,15 +136,51 @@ def state_dict(
     def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True) -> _IncompatibleKeys:
         return self._original_module.load_state_dict(state_dict=state_dict, strict=strict)
 
+    def _redirection_through_forward(self, method_name: str) -> Callable:
+        assert method_name != "forward"
+        original_forward = self._original_module.forward
+
+        def wrapped_forward(*args: Any, **kwargs: Any) -> Any:
+            # Unpatch ourselves immediately before calling the method `method_name`
+            # because itself may want to call the real `forward`
+            self._original_module.forward = original_forward
+            # Call the actual method e.g. `.training_step(...)`
+            return getattr(self._original_module, method_name)(*args, **kwargs)
```

Comment:
nit: split the getattr from the actual call for to have a better stack trace in case of errors

```suggestion
            method = getattr(self._original_module, method_name)
            return method(*args, **kwargs)
```